components/playlist/playlist_download_request_manager.cc

```diff
@@ -115,10 +115,28 @@ void PlaylistDownloadRequestManager::RunMediaDetector(Request request) {
   callback_for_current_request_ = std::move(request.callback);
   DCHECK(!callback_for_current_request_.is_null());
 
-  GURL url(request.url);
-  DCHECK(url.is_valid());
-  web_contents_->GetController().LoadURLWithParams(
-      content::NavigationController::LoadURLParams(url));
+  if (absl::holds_alternative<std::string>(request.url_or_contents)) {
+    CreateWebContents();
+    GURL url(absl::get<std::string>(request.url_or_contents));
+    DCHECK(url.is_valid());
+    DCHECK(web_contents_);
+    web_contents_->GetController().LoadURLWithParams(
+        content::NavigationController::LoadURLParams(url));
+  } else {
+    auto weak_contents =
+        absl::get<base::WeakPtr<content::WebContents>>(request.url_or_contents);
+    if (!weak_contents) {
+      FetchPendingRequest();
+      return;
+    }
+
+    DCHECK(weak_contents->GetMainFrame());
+    weak_contents->GetMainFrame()->ExecuteJavaScriptInIsolatedWorld(
+        base::UTF8ToUTF16(media_detector_script_),
+        base::BindOnce(&PlaylistDownloadRequestManager::OnGetMedia,
+                       weak_factory_.GetWeakPtr()),
+        g_playlist_javascript_world_id);
```

Comment:
How about making this as a separated method like `GetMedia(WebContents* web_contents);` and share with `DidFinishLoad()`?